govcd/vdc.go

```diff
@@ -952,29 +952,15 @@ func (vdc *Vdc) GetVAppTemplateByName(vAppTemplateName string) (*VAppTemplate, e
 	if err != nil {
 		return nil, err
 	}
-	return vdc.GetVAppTemplateByHref(vAppTemplateQueryResult.HREF)
-}
-
-// GetVAppTemplateByHref finds a vApp template by HREF
-// On success, returns a pointer to the vApp template structure and a nil error
-// On failure, returns a nil pointer and an error
-func (vdc *Vdc) GetVAppTemplateByHref(href string) (*VAppTemplate, error) {
-	return getVAppTemplateByHref(vdc.client, href)
-}
-
-// GetVAppTemplateById finds a vApp Template by ID.
-// On success, returns a pointer to the VAppTemplate structure and a nil error.
-// On failure, returns a nil pointer and an error.
-func (vdc *Vdc) GetVAppTemplateById(vAppTemplateId string) (*VAppTemplate, error) {
-	return getVAppTemplateById(vdc.client, vAppTemplateId)
+	return getVAppTemplateByHref(vdc.client, vAppTemplateQueryResult.HREF)
 }
 
 // GetVAppTemplateByNameOrId finds a vApp Template by Name or ID.
 // On success, returns a pointer to the VAppTemplate structure and a nil error
 // On failure, returns a nil pointer and an error
 func (vdc *Vdc) GetVAppTemplateByNameOrId(identifier string, refresh bool) (*VAppTemplate, error) {
 	getByName := func(name string, refresh bool) (interface{}, error) { return vdc.GetVAppTemplateByName(name) }
-	getById := func(id string, refresh bool) (interface{}, error) { return vdc.GetVAppTemplateById(id) }
+	getById := func(id string, refresh bool) (interface{}, error) { return getVAppTemplateById(vdc.client, id) }
```

Comment:
At first I had a worry that `getByName` and `getById` are misaligned (one looks at VDC scope, other at whole VCD), but given that this is a `*Vdc` receiver - should be good.

Reply:
Yep, it looks weird at first glance. But as `getVAppTemplateById` doesn't use VDC details we should be good.
Before this PR, I added `GetVAppTemplateById` to the `*Vdc` receiver as the idea was to fetch vApp Templates from Catalogs and from VDCs, but in reality, by ID one can fetch them wherever they want.